pkg/handler/handler.go

```diff
@@ -259,9 +230,31 @@ func (m *Modifier) addEnvToContainer(container *corev1.Container, tokenFilePath,
 	return changed
 }
 
-func (m *Modifier) updatePodSpec(pod *corev1.Pod, roleName, audience string, regionalSTS bool, tokenExpiration int64) ([]patchOperation, bool) {
-	updateSettings := newPodUpdateSettings(m.AnnotationDomain, pod, regionalSTS)
+// parsePodAnnotations parses the pod annotations that can influence mutation:
+// - tokenExpiration. Overrides the given service account annotation/flag-level
```

Comment:
Nit: this ship probably already sailed, but should the service account annotation for token expiration really be overridable by the pod?  I.e. should someone with only permissions to the pod, but not the service account, be allowed to increase the expiration (vs. only being allowed to lower it)?

Reply:
they already can by manually editing the pod. but the whole point of the webhook is so you dont have to manually edit anything.

I get what youre saying though in the sense that I dont really understand the logic why this in particular is controlled at pod level whereas the rest are serviceaccount/flag. at least with the comments it should be easier to navigate. because as it stands there are basically 4 * flags + 4 * sa annotations + 1 * pod annotation